### PR TITLE
feat(coder): LibreChat Coder agent — converse-delegated subagent + authenticated port-share (P1/P2 applied)

### PR DESCRIPTION
**Pull request by Guy-Ghis**
## 1. Summary

This PR implements the **LibreChat Coder agent (Ticket #832)** following the **converse→coder subagent-delegation** direction (ADR-0086/0088), and applies the maintainer's review feedback (P1 + P2s).

Changes:

- Registers the `coder` DB agent in `charts/librechat-app/values.yaml` `agentSeed.agents`, delegated by `converse` via `subagentNames` (idempotent `seed-agents.js`).
- Resolvable ids: `model: adorsys-coder-pro-internal` + `mcpServers: ["coder_mcp"]` (→ `sys__all__sys_mcp_coder_mcp`), both verified against the converse-provider model list + registered MCP servers.
- Port-sharing contract per ADR-0121.
- Restores `docs/adr/0086-...md` and `docs/integrations/coder-workspace-urls.md` to `origin/main` (the prior prompt-centric PR rewrote the ADR + deleted ~537 lines of hard-won Coder knowledge).

### Reviewer feedback applied (PR #937)
- **P1 (block):** HARD-set the `coder` agent to `share_level: authenticated` only. It must never create a `public` (unauthenticated) share — public exposure requires a human admin decision and is outside the agent's authority. Applied to the coder prompt (chart values), the live seed, and the workspace-urls contract.
- **P2a:** `charts/librechat-app/values.yaml` `agentSeed.agents` is the **canonical** fleet source; ADR-0086 + system-prompt doc reference it instead of re-declaring.
- **P2b:** documented the `coder_mcp` credential scope (shared deployment-level LibreChat OAuth client; per-user isolation via `X-USER-*` headers; keep non-admin/workspace-scoped; never an admin/org-wide Coder token).
- **P2c:** added `tools/check-agent-seed.sh` validating every `agentSeed.agents[*]` model id + `mcpServers[*]` → `sys__all__sys_mcp_<name>` token expansion.

It solves:

- https://github.com/ADORSYS-GIS/ai-helm/issues/832 (LibreChat Coder agent) — parent Epic #821

---

## 2. Intent

> Enable web-application scaffolding from LibreChat by giving the `coder` agent (delegated from `converse`) a secure, resolvable Coder integration that returns a live preview link.

---

### In Scope

- `coder` subagent under `converse`, `agentSeed.agents`, resolvable model/MCP ids.
- Port-share contract (authenticated-only).
- ADR-0086 + `coder-workspace-urls.md` restoration.
- Reviewer P1 + P2 fixes (authenticated-only, canonical roster, credential scope, seed-resolve check).

### Out of Scope

- Coder cluster Helm charts / server deployment (`charts/coder`).
- OpenCode runtime binaries inside workspaces.
- Live e2e verification of the scaffolded-app flow (outstanding).

---

## 4. Verification

I verified this change by:

- [x] Running automated tests
- [x] Running manual tests
- [ ] Checking logs
- [ ] Checking metrics
- [x] Testing error cases
- [x] Testing permissions/security behavior
- [ ] Testing rollback or failure behavior, if relevant

Commands run:

```bash
helm lint charts/librechat-app
helm template test-release charts/librechat-app --set agentSeed.enabled=true --set agentSeed.platformUserEmail=platform@ai.camer.digital
PATH=<mikefarah-yq-bin>:$PATH ./tools/check-agent-seed.sh
```

Results:

```text
==> Linting charts/librechat-app
1 chart(s) linted, 0 chart(s) failed
# helm template renders; seed contains model adorsys-coder-pro-internal +
# mcpServers [coder_mcp]
# check-agent-seed: OK (validates converse + coder agent model/mcp ids)
```

---

## 5. Screenshots / Evidence

* `tools/check-agent-seed.sh` — seed resolve-check.
* Design: ADR-0086 (`docs/adr/0086-librechat-agent-fleet-and-gitops-seed.md`), ADR-0121, [`coder-agent-system-prompt.md`](docs/integrations/coder-agent-system-prompt.md).

---

## 6. Risk Assessment

Risk level:

* [x] Low
* [ ] Medium
* [ ] High

Potential risks:

* A wrong model/mcp id seeds a broken agent at PostSync.

Mitigation:

* Values reconciled against the converse model list + registered MCP; `check-agent-seed.sh` CI guard; P1 hard-limits `public` shares.

---

## 7. AI Usage Declaration

AI was used for:

* [x] Understanding existing code
* [x] Generating code
* [ ] Refactoring
* [ ] Generating tests
* [x] Drafting documentation
* [x] Reviewing the diff
* [ ] Not used

Human verification:

* [x] I understand every meaningful change in this PR
* [x] I checked generated code manually
* [x] I checked generated tests manually
* [x] I removed unsupported AI assumptions
* [x] I accept responsibility for this PR

---

## 8. Reviewer Focus

Please focus your review on:

* [x] Correctness
* [x] Architecture
* [x] Security
* [ ] Performance
* [ ] Tests
* [x] Maintainability
* [x] Product intent
* [ ] Edge cases

### CODE REVIEW

**Comment by Guy-Ghis** [comment]
### Update: Refactored to Subagent Delegation Pattern

Per architectural feedback, we have updated PR #937 to configure the **Coder Agent (`coder`)** as a specialized **Subagent** delegated to by primary orchestrators (`converse`) rather than a standalone top-level persona:

1. **Fleet Seed Configuration (`charts/librechat-app/values.yaml`):**
   - Configured `coder` with `category: "subagent"`.
   - Updated `converse` agent definition with `subagentNames: ["coder"]`.
   - Preserved full system prompt, Coder REST API port sharing rules (`POST /api/v2/workspaces/{id}/port-share`), and RFC 1035 limits within the `coder` subagent instructions.

2. **Integration Documentation:**
   - Updated `docs/integrations/coder-agent-system-prompt.md` and `docs/integrations/coder-workspace-urls.md` to document the subagent delegation pattern and ADR-0086 compliance.

Commit: `6ac69cf31247e7257085a7a4357288b76ecb75ed`

File: charts/librechat-app/values.yaml [hunk 1]
```diff
@@ -114,7 +114,61 @@ agentSeed:
   # Override only if the Service name/namespace differ from the defaults above.
   librechatUrl: ""
   image: ""
-  agents: []
```
Comment (stephane-segning) [inline charts/librechat-app/values.yaml:126]: These agents are normally managed on ai-helm-values.

**Review by stephane-segning — CHANGES_REQUESTED** [review]

File: charts/librechat-app/values.yaml (hunk 1 above)
Comment (Guy-Ghis) [reply charts/librechat-app/values.yaml:126]: Agreed — fixed. The agent fleet (incl. the `converse` + `coder` subagents) is now removed from this chart's default: `charts/librechat-app/values.yaml` `agentSeed.agents` is `[]` (structural defaults only). The live fleet + prompt config live in `ai-helm-values` (`environments/<env>/values/librechat-app.yaml` `agentSeed.agents`, enabled: true) per ADR-0055/56 values-repo-first, pushed in `ai-helm-values` PR #220. So the fleet has a single source of truth and is no longer split/duplicated across the two repos. (Commit: `3e18d302`.)
